--- man-db-2.6.1/src/ult_src.c.orig	2012-02-05 14:21:24.000000000 +0100
+++ man-db-2.6.1/src/ult_src.c	2012-03-20 18:00:09.000000000 +0100
@@ -343,6 +343,14 @@
 				free (base);
 				base = appendstr (NULL, path, "/", include,
 						  NULL);
+				if (strchr(include, '/') == NULL) {
+					char *sec = strrchr(include, '.');
+					debug ("ult_src: wrong .so include %s\n", buffer);
+					if (sec != NULL && *sec++ != '\0') {
+						free (base);
+						base = appendstr (NULL, path, "/man", sec, "/", include, NULL);
+					}
+				}
 				free (include);
 
 				debug ("ult_src: points to %s\n", base);
